src/Reduce.php

```diff
@@ -32,37 +34,49 @@ public static function toValue(iterable $data, callable $reducer, $initialValue
     /**
      * Reduces given iterable to its min value.
      *
-     * Items of given collection must be comparable.
+     * If comparator is null then items of given collection must be comparable.
      *
      * Returns null if given collection is empty.
      *
      * @param iterable<mixed> $data
+     * @param callable|null   $comparator
      *
      * @return mixed|null
      */
-    public static function toMin(iterable $data)
+    public static function toMin(iterable $data, ?callable $comparator = null)
```

Comment:
For optional values, you wouldn't want to make it nullable. Nullable means null is a valid value to pass in, which would be wrong. Optional means that the parameter can be left off, in which case it will initialize to null. So you can write this without the `?` like:

```php
public static function toMin(iterable $data, callable $comparator = null)
```

However, PHP is a bit strange and I think you will still be able to pass null in when it defaults to null. But if the optional default is something other than null, then PHP would throw a TypeError if you pass in null. In general, you want to avoid allowing null as a valid value in an interface, especially for new code. For the optional default it is fine because it doesn't leak into the client code.

Reply:
Done.